nexuse2e-webapp/src/main/frontend/src/app/data/data.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { environment } from "../../environments/environment";
+
+@Injectable({
+  providedIn: "root",
+})
+export class DataService {
+  API_URL: string;
+
+  constructor(private http: HttpClient) {
+    if (environment.API_URL.length) {
+      this.API_URL = environment.API_URL;
+    } else {
+      this.API_URL = "../api";
+    }
+  }
+
+  get<T>(path: string): Promise<T> {
```

Comment:
We will probably have a lot of methods in the backend later on and it would be time-consuming to build the methods in the frontend accordingly.

Reply:
Yes this would be some work, but this would be only typing work. But in this way you give up typesafety and making refactoring harder. If you need to change a method or want to remove a server function you can relay on the compiler for checking.

Reply:
Makes sense, we will create a separate PR for this change.